presto-main/src/main/java/com/facebook/presto/server/AsyncHttpExecutionMBean.java

```diff
@@ -28,15 +28,15 @@
 public class AsyncHttpExecutionMBean
 {
     private final ThreadPoolExecutorMBean responseExecutor;
-    private final ThreadPoolExecutorMBean timeoutExecutor;
+    private final ScheduledExecutorService timeoutExecutor;
```

Comment:
The `ScheduledExecutorService` does not expose any JMX properties by itself. Instead some kind of `ConcurrentScheduledExecutorMBean` has to be implemented. The `ConcurrentScheduledExecutorMBean` should expose `ConcurrentScheduledExecutor` specific statistics, similar to how the `ThreadPoolExecutorMBean` exposes the stats for `ThreadPoolExecutor`.

We can do this as a follow up. As for now i would rather prefer the  `timeoutExecutor` to be completely removed. As since it is does not provide any MXBean stats keeping it here is only misleading.

Reply:
Agreed. Sounds like a good follow-up :)